### PR TITLE
feat: add Intro Screen

### PR DESCRIPTION
**Pull request by DylanRJ**
# DCMAW-6672: iOS | UI Frameworks | Create common package code for the Welcome screen

This PR adds a Welcome screen to this package. An icon, title, body and button are the screen elements and the functionality is driven through the `viewModel` property which is used extensively in the view controller.

# Checklist

## Before raising your pull request:
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with ticket ID and a short description about the feature or update
      i.e. _DCMAW-222: Added ReadID SDK to iOS app_
- [x] Created a `draft` pull request if it is not yet ready for review

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

- [x] Met all accessibility requirements?
    - [x] Checked dynamic type sizes are applied
    - [x] Checked VoiceOver can navigate your new code
    - [x] Checked a user can navigate only using a keyboard around your new code 

## Before merging your pull request:
- [x] Ensure that the code coverage and SonarCloud checks have passed
- [x] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
~- [ ] Ran the app to ensure that no regressions have been caused by changes during code review.~
~- [ ] Targeted the correct branch; `develop`, `release` or `main`~

### CODE REVIEW

File: Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/WelcomeView.xib
Comment (mcgarj) [inline Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/WelcomeView.xib:1]: The Scroll View in the storyboard is giving a warning of missing constraints, do these need to be added?

Comment (DylanRJ) [reply Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/WelcomeView.xib:1]: I brought this storyboard file directly over from the [ID Check project](https://github.com/alphagov/di-ipv-dca-mob-ios) and didn't change any of the constraints so shouldn't be any different from what we have in the live app at the monent.
Maybe @benmckillop would be able to check if those constraints are okay when he gets a moment 🙂

Comment (benmckillop) [reply Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/WelcomeView.xib:1]: @DylanRJ if we wait until my PR is merged, you can add this screen to the demo project and ensure it works as expected :)

Comment (robinphillips) [reply Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/WelcomeView.xib:1]: I've had a go at fixing this. Would be good to have a chat before I commit / push :)

**Review by benmckillop — APPROVED** [review]
Looks good to me, providing you have chatted to @robinphillips around the comment about constraints :)

File: GDSCommon-Demo/GDSCommon-Demo/Mocks/MockIntroViewModel.swift
Comment (robinphillips) [inline GDSCommon-Demo/GDSCommon-Demo/Mocks/MockIntroViewModel.swift:6]: Think this mock text should be updated to `intro`

Comment (DylanRJ) [reply GDSCommon-Demo/GDSCommon-Demo/Mocks/MockIntroViewModel.swift]: Done 🙂

**Review by robinphillips — APPROVED** [review]

**Pull request merged**
